spec/konfa_spec.rb

```diff
@@ -369,4 +360,82 @@ def allowed_variables
       expect(MyOtherTestKonfa.get :my_var).to eq 'true'
     end
   end
+
+
+  context "#read_from" do
+    subject { MyTestKonfa.read_from(initializer, good_file) }
+
+    context 'valid initializer' do
+      let(:initializer) { :yaml }
+
+      it { is_expected.to eq MyTestKonfa }
+
+      it 'calls the initializer' do
+        expect(MyTestKonfa).to receive(:init_with_yaml).with(good_file)
+        subject
+      end
+    end
+
+    context 'invalid initializer' do
+      let(:initializer) { :foobar }
+
+      it 'raises error' do
+        expect {
+          subject
+        }.to raise_error(Konfa::UnsupportedInitializerError)
+      end
+    end
+  end
+
+  context '#initialized!' do
+    subject { MyTestKonfa.initialized! }
+
+    it 'sets initialized to true' do
+      expect {
+        subject
+      }.to change { MyTestKonfa.initialized? }.from(false).to(true)
+    end
+
+    it 'calls after_initialize' do
+      expect(MyTestKonfa).to receive(:after_initialize)
+      subject
+    end
+
+    it 'returns self' do
+      expect(subject).to eq MyTestKonfa
+    end
+
+    context 'when already initialized' do
+      before(:each) { allow(MyTestKonfa).to receive(:initialized?).and_return(true) }
+
+      it 'does not run' do
+        expect(MyTestKonfa).not_to receive(:after_initialize)
+        expect(subject).to be_nil
```

Comment:
Should it not return self in this case also? Or return a boolean indicating whether it ran or not

Reply:
It would normally return `self`, so `nil` is an indicator that it didn't run. But maybe `false` is more meaningful? Feels kinda awkward to return `self` or `false` though

Reply:
If the return value should indicate success or not, I think it should return a boolean. If it should be used to chain calls, it should return self. If it is not significant to try to initialize twice, I think we can return self and leave the responsibility of checking status to the consumer code.